src/com/esotericsoftware/kryo/Kryo.java

```diff
@@ -374,6 +374,10 @@ public Serializer getDefaultSerializer (Class type) {
 		Serializer serializerForAnnotation = getDefaultSerializerForAnnotatedType(type);
 		if (serializerForAnnotation != null) return serializerForAnnotation;
 
+		if (type.isAssignableFrom(KryoSerializable.class) && type != Object.class) {
+			return new KryoSerializableSerializer();
+		}

```

Comment:
Why do we need this condition? Shouldn't the subsequent loop return a `KryoSerializableSerializer` anyway? And why the `type != Object.class`?

Reply:
Yes, you're right. This was wrong.

Reply:
I simplified the fix by only moving the KryoSerializable to the end of the list.